test/e2e/storage/testsuites/provisioning.go

```diff
@@ -239,6 +243,156 @@ func (p *provisioningTestSuite) DefineTests(driver storageframework.TestDriver,
 		l.testCase.TestDynamicProvisioning()
 	})
 
+	ginkgo.It("should provision storage with any volume data source [Feature:AnyVolumeDataSource]", func() {
+		if len(dInfo.InTreePluginName) != 0 {
+			e2eskipper.Skipf("AnyVolumeDataSource feature only works with CSI drivers - skipping")
+		}
+		if pattern.VolMode == v1.PersistentVolumeBlock {
+			e2eskipper.Skipf("Test for Block volumes is not implemented - skipping")
+		}
+
+		init()
+		defer cleanup()
+
+		ginkgo.By("Creating populator namespace")
+		popNamespace, err := f.CreateNamespace(f.Namespace.Name, map[string]string{
+			"e2e-framework":      f.BaseName,
+			"e2e-test-namespace": f.Namespace.Name,
+		})
+		framework.ExpectNoError(err)
+
+		defer func() {
+			f.DeleteNamespace(popNamespace.Name)
+		}()
+
+		ginkgo.By("Deploying hello-populator")
```

Comment:
so is this test dependent on the hello type which it does not install and is not in core?

generally not a fan of adding e2e tests that are never going to be portable.

if there are core kubernetes features / code changes / ... related to this, we should come up with a portable way to test them. if it's entirely out-of-tree, we should test it out-of-tree.

Reply:
The intention of the test is not to test the example populator itself, but to test that one of the populators actually works. But I agree with your point on portability, so let me think about it if there is a good way.